### PR TITLE
CoW: Warn for cases that go through putmask

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Open for suggestions how we can improve the put mask mechanism to avoid duplicate warnings

xref https://github.com/pandas-dev/pandas/issues/56019

### CODE REVIEW

File: pandas/core/internals/base.py [hunk 1]
```diff
@@ -49,6 +52,11 @@
     )
 
 
+class _AlreadyWarned:
+    def __init__(self):
+        self.warned_already = False
```
Comment (jorisvandenbossche) [inline pandas/core/internals/base.py:57]: Can you add a small comment here about why such a class is needed (in contrast to a simple boolean argument)?

**Review by jorisvandenbossche — COMMENTED** [review]
With https://github.com/pandas-dev/pandas/pull/55522, we will also have to pass down whether we already warned about chained assignment

File: pandas/core/internals/base.py (hunk 1 above)
Comment (phofl) [reply pandas/core/internals/base.py]: added

**Comment by jorisvandenbossche** [comment]
> With https://github.com/pandas-dev/pandas/pull/55522, we will also have to pass down whether we already warned about chained assignment

I added a small commit to do this, the other comments we can address in follow-ups

**Pull request merged**